### PR TITLE
Added CHANGELOG for 1.12.2 release

### PR DESCRIPTION
**Pull request by philippjfr**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,41 @@
+Version 1.12.2
+==============
+
+This is a minor release with a number of important bug fixes.
+
+Enhancements:
+
+- Add styles argument to draw tool streams to allow cycling colors
+  when drawing glyphs
```
Comment (jbednar) [inline CHANGELOG.md]:
```suggestion
  and other styling when drawing glyphs
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+- Ensure that bokeh server plot updates have the exclusive Document
+  lock ([#3621](https://github.com/pyviz/holoviews/pull/3621))
+- Ensure that Dimensioned streams are inherited on `__mul__`
+  ([#3658](https://github.com/pyviz/holoviews/pull/3658))
+- Ensure that bokeh hover tooltips are updated when dimensions change
+  ([#3609](https://github.com/pyviz/holoviews/pull/3609))
+- Fix DynamicMap.event method for empty streams
+  ([#3564](https://github.com/pyviz/holoviews/pull/3564))
+- Ensure that color cycles on Violin and BoxWhisker elements
+  ([#3592](https://github.com/pyviz/holoviews/pull/3592))
```
Comment (jbednar) [inline CHANGELOG.md]: Shouldn't 3592 go under Compatibility, as it's a backwards-incompatible change to the defaults? (If I understood the original issue correctly.)

Comment (philippjfr) [reply CHANGELOG.md]: Sure, or maybe it's own section since I intended Compatibility sections to be about compatibility with other libraries.

Comment (jbednar) [reply CHANGELOG.md]: As you wish!

**Pull request merged**